reporting/CassandraBackend.cpp

```diff
@@ -501,68 +501,128 @@ CassandraBackend::fetchBookOffers(
     std::uint32_t limit,
     std::optional<ripple::uint256> const& cursor) const
 {
-    auto index = getBookIndexOfSeq(sequence);
-    if (!index)
-        return {};
-    BOOST_LOG_TRIVIAL(info) << __func__ << " index = " << std::to_string(*index)
-                            << " book = " << ripple::strHex(book);
-    BookOffersPage page;
-    ripple::uint256 zero = {};
+    auto rng = fetchLedgerRange();
+    
+    if(!rng)
+        return {{},{}};
+
+    uint32_t upper = sequence;
+    auto lastPage = rng->maxSequence - (rng->maxSequence % 256);
+
+    auto readBooks = [this](
+            ripple::uint256 const& book, 
+            std::uint32_t seq,
+            std::optional<ripple::uint256> const& cursor) 
+            -> std::vector<std::pair<std::uint64_t, ripple::uint256>>
     {
-        CassandraStatement statement{selectBook_};
-        statement.bindBytes(zero);
-        statement.bindInt(*index);
-        statement.bindBytes(zero);
-        statement.bindUInt(1);
-        CassandraResult result = executeSyncRead(statement);
-        if (!result)
-            page.warning = "Data may be incomplete";
+        CassandraStatement statement{this->selectBook_};
+        std::vector<std::pair<std::uint64_t, ripple::uint256>> keys = {};
+    
+        statement.bindBytes(book.data(), 24);
+        statement.bindInt(seq);
+
+        BOOST_LOG_TRIVIAL(info) << __func__ << " upper = " << std::to_string(seq)
+                                << " book = " << ripple::strHex(std::string((char*)book.data(), 24));
+
+        if (cursor)
+        {
+            auto object = this->fetchLedgerObject(*cursor, seq);
+
+            if(!object)
+                return {{}, {}};
+
+            ripple::SerialIter it{object->data(), object->size()};
+            ripple::SLE offer{it, *cursor};
+            ripple::uint256 bookDir = offer.getFieldH256(ripple::sfBookDirectory);
+
+            statement.bindBytes(bookDir.data() + 24, 8);
+            statement.bindBytes(*cursor);
+        }
```

Comment:
Let's remove this, since we are punting on cursors for book_offers for now. You can keep the query the same, and just bind 0, like the else clause does.